R/gm-auth.R

```diff
@@ -0,0 +1,207 @@
+# This file is the interface between gmailr and the
+# auth functionality in gargle.
+.auth <- gargle::init_AuthState(
+  package     = "gmailr",
+  auth_active = TRUE # TODO: what is this?
```

Comment:
@jennybc can you explain this and verify this is the correct setting for gmailr. I think it is because the Gmail API does not accept token free requests.

Reply:
`auth_active = TRUE` means "send a token with requests"
`auth_active = FALSE` means "send no token and send an API key with requests"

Correct, AFAICT, `auth_active` must be `TRUE` for gmailr at all times, as is the case for bigrquery. So the fact that this can be toggled is irrelevant to you. But is useful in other packages.